components/utils/src/templates.rs

```diff
@@ -25,23 +25,8 @@ pub fn render_template(
     context: Context,
     theme: &Option<String>,
 ) -> Result<String> {
-    // check if it is in the templates
-    if tera.templates.contains_key(name) {
-        return tera.render(name, &context).map_err(std::convert::Into::into);
-    }
-
-    // check if it is part of a theme
-    if let Some(ref t) = *theme {
-        let theme_template_name = format!("{}/templates/{}", t, name);
-        if tera.templates.contains_key(&theme_template_name) {
-            return tera.render(&theme_template_name, &context).map_err(std::convert::Into::into);
-        }
-    }
-
-    // check if it is part of ZOLA_TERA defaults
-    let default_name = format!("__zola_builtins/{}", name);
-    if tera.templates.contains_key(&default_name) {
-        return tera.render(&default_name, &context).map_err(std::convert::Into::into);
+    if let Some(template) = check_template_fallbacks(name, tera, theme) {
```

Comment:
I'm wondering if adding an optional `fallback` param to that function would simplify things, so you don't have to call `check_template_fallbacks` manually would be worth it. What do you think?

Reply:
That does sound like a good idea. Would you envisage passing an `Option<&str>` as the fallback and just running the tests twice? It would be nice to avoid duplicating the code, but aside from making a `Vec<&str>` and iterating over that I can't think of a way of doing so off the top of my head.

Reply:
I'm trying to remember what I had in mind... but I think it was `Option<&str>`

Reply:
Having a go at that now. If we implement a fallback would we throw and error when constructing the Paginator?
Thinking of these lines in components/library/src/pagination/mod.rs:
```
// Check for taxon-specific template, or use generic as fallback.
let specific_template = format!("{}/single.html", taxonomy.kind.name);
let template = match check_template_fallbacks(&specific_template, tera, theme) {
    Some(template) => template,
    None => "taxonomy_single.html",
};
```
With a fallback param check_template_fallbacks will now presumably always return the template if it exists. Currently paginated taxonomies only error when attempting to render. I could convert the match into an unwrap, but that would move the location where the logic fails.

Sorry to be a pain, just don't want to do something that might make later re-factoring more difficult.